pkg/controller/networkpolicy/clustergroup.go

```diff
@@ -49,10 +50,19 @@ func (n *NetworkPolicyController) updateClusterGroup(oldObj, curObj interface{})
 	klog.V(2).Infof("Processing UPDATE event for ClusterGroup %s", cg.Name)
 	newGroup := n.processClusterGroup(cg)
 	oldGroup := n.processClusterGroup(og)
+	selectorUpdated := getNormalizedNameForSelector(newGroup.Selector) != getNormalizedNameForSelector(oldGroup.Selector)
 	ipBlockUpdated := newGroup.IPBlock != oldGroup.IPBlock
 	svcRefUpdated := newGroup.ServiceReference != oldGroup.ServiceReference
-	if getNormalizedNameForSelector(newGroup.Selector) == getNormalizedNameForSelector(oldGroup.Selector) && !ipBlockUpdated && !svcRefUpdated {
-		// No change in the selectors of the ClusterGroup. No need to enqueue for further sync.
+	oldChildGroups, newChildGroups := sets.String{}, sets.String{}
+	for _, c := range oldGroup.ChildGroups {
+		oldChildGroups.Insert(c)
+	}
+	for _, c := range newGroup.ChildGroups {
+		newChildGroups.Insert(c)
+	}
+	childGroupsUpdated := !oldChildGroups.Equal(newChildGroups)
+	if !selectorUpdated && !ipBlockUpdated && !svcRefUpdated && !childGroupsUpdated {
```

Comment:
Better to keep the original optimization here - if one condition is met, no need to calculate the following ones, esp. childGroupsUpdated seems not very cheap to compute.

Reply:
They should all be fairly cheap (one-pass string set comparisons only), but refactored